presto-main/src/main/java/io/prestosql/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -697,11 +698,25 @@ public Void visitTableScan(TableScanNode node, Void context)
         public Void visitValues(ValuesNode node, Void context)
         {
             NodeRepresentation nodeOutput = addNode(node, "Values");
-            for (List<Expression> row : node.getRows()) {
-                nodeOutput.appendDetailsLine(row.stream()
-                        .map(PlanPrinter::unresolveFunctions)
-                        .map(Expression::toString)
-                        .collect(joining(", ", "(", ")")));
+            if (node.getRows().isEmpty()) {
+                for (int i = 0; i < node.getRowCount(); i++) {
+                    nodeOutput.appendDetailsLine("()");
+                }
+                return null;
+            }
+            List<String> rows = node.getRows().get().stream()
+                    .map(row -> {
+                        if (row instanceof Row) {
+                            return ((Row) row).getItems().stream()
+                                    .map(PlanPrinter::unresolveFunctions)
+                                    .map(Expression::toString)
+                                    .collect(joining(", ", "(", ")"));
+                        }
```

Comment:
Why is this needed? The expression printer should be able to print a `Row` expression directly.

Reply:
I wanted to preserve the result for all "regular" rows.
`Row` expression renders as "ROW(..."